s3connector/s3.go

```diff
@@ -113,6 +112,16 @@ func (s *s3ObjectStorage) DeleteObject(objPath S3ObjectPath) error {
 }
 
 func (s *s3ObjectStorage) GetObject(objPath S3ObjectPath) (io.ReadCloser, error) {
+
+	var objRange *string
+
+	if objPath.Range != nil {
+		if objPath.Range.Start < 0 || objPath.Range.End <= objPath.Range.Start {
+			return nil, fmt.Errorf("invalid range: start must be non-negative and end must be greater than start")
```

Comment:
maybe add the range as well so we can understand the issue when we see the error